bundles/org.eclipse.equinox.p2.transport.ecf/src/org/eclipse/equinox/internal/p2/transport/ecf/RepositoryTransport.java

```diff
@@ -43,7 +44,30 @@
 public class RepositoryTransport extends Transport {
 
 	public static final String TIMEOUT_RETRY = "org.eclipse.equinox.p2.transport.ecf.retry"; //$NON-NLS-1$
-	private static Map<URI, Integer> socketExceptionRetry = null;
+
+	private static class Retry {
+		// 10 minutes.
+		long LIFETIME = 10 * 60 * 1000;
+		long expiration;
+		int count;
+
+		public Retry() {
+			expiration = System.currentTimeMillis() + LIFETIME;
+		}
+
+		int increment() {
```

Comment:
```suggestion
		synchronized int increment() {
```
Otherwise this is not concurrency safe!